vyper/builtin_functions/functions.py

```diff
@@ -1666,15 +1756,19 @@ def __repr__(self):
         return f"builtin function unsafe_{self.op}"
 
     def fetch_call_return(self, node):
-        validate_call_args(node, 2)
+        return_type = self.infer_arg_types(node).pop()
+        return return_type
 
+    def infer_arg_types(self, node):
+        validate_call_args(node, 2)
```

Comment:
going forward it might be nice to break out this logic into something like, `validate_binop_types`. but might be better to address in https://github.com/vyperlang/vyper/pull/2832